### ISSUE TITLE
DROID doesn't clean up BeanFactory state after run?

### ISSUE THREAD

**Issue by mistydemeo** [digital-preservation/droid#31]
DROID appears to leave the Spring framework BeanFactory in an invalid state after running once, making it impossible to launch further DROID sessions in the same JVM.

This normally isn't an issue since the JVM is shut down when DROID quits, but if DROID is launched via a persistent JVM like [nailgun](http://www.martiansoftware.com/nailgun/), this makes it impossible to run DROID again without restarting the JVM.

Backtrace:

```
java.lang.IllegalStateException: BeanFactory not initialized or already closed - call 'refresh' before accessing beans via the ApplicationContext
    at org.springframework.context.support.AbstractRefreshableApplicationContext.getBeanFactory(AbstractRefreshableApplicationContext.java:153)
    at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:880)
    at uk.gov.nationalarchives.droid.command.context.SpringUiContext.getNoProfileRunCommand(SpringUiContext.java:107)
    at uk.gov.nationalarchives.droid.command.action.CommandFactoryImpl.getNoProfileCommand(CommandFactoryImpl.java:238)
    at uk.gov.nationalarchives.droid.command.action.CommandLineParam$14.getCommand(CommandLineParam.java:167)
    at uk.gov.nationalarchives.droid.command.DroidCommandLine.run(DroidCommandLine.java:137)
    at uk.gov.nationalarchives.droid.command.DroidCommandLine.processExecution(DroidCommandLine.java:193)
    at uk.gov.nationalarchives.droid.command.DroidCommandLine.main(DroidCommandLine.java:168)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:616)
    at com.martiansoftware.nailgun.NGSession.run(Unknown Source)
```

**Comment by rhubner** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [#31 DROID doesn't clean up BeanFactory state after run](https://trello.com/c/Gw5EqKFh/12-31-droid-doesn-t-clean-up-beanfactory-state-after-run)

**Comment by DavidUnderdown** [comment]
In test

**Comment by rhubner** [comment]
Fixed in 6.4.
